src/witan/models/dem/ccm/core/projection_loop.clj

```diff
@@ -12,48 +12,67 @@
             [witan.models.dem.ccm.models-utils :as m-utils]
             [taoensso.timbre :as timbre]))
 
+(defn create-empty-ds
+  [schema]
+  (ds/dataset (map (comp :v :schema) (:column-names schema)) []))
+
 (defworkflowpred finished-looping?
   {:witan/name :ccm-core/ccm-loop-pred
    :witan/version "1.0.0"
-   :witan/input-schema {:population PopulationSchema :loop-year (s/constrained s/Int m-utils/year?)}
-   :witan/param-schema {:last-proj-year (s/constrained s/Int m-utils/year?)}
+   :witan/input-schema {:population PopulationSchema}
+   :witan/param-schema {:last-proj-year YearSchema}
    :witan/exported? true}
-  [{:keys [loop-year]} {:keys [last-proj-year]}]
-  (>= loop-year last-proj-year))
+  [{:keys [population]} {:keys [last-proj-year]}]
+  (let [loop-year (m-utils/get-last-year population)]
+    (>= loop-year last-proj-year)))
 
-(defworkflowfn prepare-inputs
+(defworkflowfn prepare-inputs-1-0-0
   "Step happening before the projection loop.
    Takes in the historic population and outputs the latest year and
    the population for that year which will be updated within the
    projection loop. Also outputs the the population that will ultimately
    contain the historic population and the population projections."
```

Comment:
doc strings need updating generally but have created an epic which includes this